pinot-core/src/main/java/org/apache/pinot/core/util/OsCheck.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import java.util.Locale;
+
+
+public final class OsCheck {
+
+  // cached result of OS detection
+  protected static OSType _detectedOS;
+
+  private OsCheck() {
+  }
+
+  /**
+   * detect the operating system from the os.name System property and cache
+   * the result
+   *
+   * @return - the operating system detected
+   */
+  public static OSType getOperatingSystemType() {
+    if (_detectedOS == null) {
+      String os = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
+      if ((os.contains("mac")) || (os.contains("darwin"))) {
+        _detectedOS = OSType.MacOS;
+      } else if (os.contains("win")) {
+        _detectedOS = OSType.Windows;
+      } else if (os.contains("nux")) {
```

Comment:
I've never seen "nux" used for this before, what does it capture which "linux" wouldn't?

Reply:
changed to "linux"